be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -890,6 +890,10 @@ Status BetaRowsetWriter::_flush_segment_writer(std::unique_ptr<segment_v2::Segme
     segstat.index_size = index_size + (*writer)->get_inverted_index_file_size();
     segstat.key_bounds = key_bounds;
 
+    if (_delta_writer != nullptr) {
+        _delta_writer->update_filtered_rows((*writer)->num_rows_filtered());
```

Comment:
You should store this value in rowset writer and get it from delta writer instead.
i.e. Call `rowset_writer->get_num_rows_filtered()` in DeltaWriter.